diff -ruN openssl-0.9.8h.orig/Configure openssl-0.9.8h/Configure
--- openssl-0.9.8h.orig/Configure	2008-05-01 19:11:30.000000000 -0400
+++ openssl-0.9.8h/Configure	2008-05-30 09:34:10.000000000 -0400
@@ -515,6 +515,9 @@
 "darwin64-x86_64-cc","cc:-arch x86_64 -O3 -fomit-frame-pointer -DL_ENDIAN -DMD32_REG_T=int -Wall::-D_REENTRANT:MACOSX:-Wl,-search_paths_first%:SIXTY_FOUR_BIT_LONG RC4_CHAR RC4_CHUNK BF_PTR2 DES_INT DES_UNROLL:${no_asm}:dlfcn:darwin-shared:-fPIC -fno-common:-arch x86_64 -dynamiclib:.\$(SHLIB_MAJOR).\$(SHLIB_MINOR).dylib",
 "debug-darwin-ppc-cc","cc:-DBN_DEBUG -DREF_CHECK -DCONF_DEBUG -DCRYPTO_MDEBUG -DB_ENDIAN -g -Wall -O::-D_REENTRANT:MACOSX::BN_LLONG RC4_CHAR RC4_CHUNK DES_UNROLL BF_PTR::osx_ppc32.o::::::::::dlfcn:darwin-shared:-fPIC -fno-common:-dynamiclib:.\$(SHLIB_MAJOR).\$(SHLIB_MINOR).dylib",
 
+"darwin-i386-iphonesimulator-gcc","/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin/gcc-4.2:-arch i386 -isysroot <SDKROOT> -O3 -fomit-frame-pointer -fno-common::-D_REENTRANT:MACOSX:-Wl,-search_paths_first%:BN_LLONG RC4_CHAR RC4_CHUNK DES_UNROLL BF_PTR:${no_asm}:dlfcn:darwin-shared:-fPIC -fno-common:-arch i386 -dynamiclib:.\$(SHLIB_MAJOR).\$(SHLIB_MINOR).dylib",
+"darwin-i386-iphoneos-gcc","/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/gcc-4.2:-arch armv6 -isysroot <SDKROOT> -O3 -fomit-frame-pointer -fno-common::-D_REENTRANT:MACOSX:-Wl,-search_paths_first%:BN_LLONG RC4_CHAR RC4_CHUNK DES_UNROLL BF_PTR:${no_asm}:dlfcn:darwin-shared:-fPIC -fno-common:-arch armv6 -dynamiclib:.\$(SHLIB_MAJOR).\$(SHLIB_MINOR).dylib",
+
 ##### A/UX
 "aux3-gcc","gcc:-O2 -DTERMIO::(unknown):AUX:-lbsd:RC4_CHAR RC4_CHUNK DES_UNROLL BF_PTR:::",
 
diff -ruN openssl-0.9.8h.orig/crypto/ui/ui_openssl.c openssl-0.9.8h/crypto/ui/ui_openssl.c
--- openssl-0.9.8h.orig/crypto/ui/ui_openssl.c	2005-09-25 22:40:18.000000000 -0400
+++ openssl-0.9.8h/crypto/ui/ui_openssl.c	2008-05-30 09:34:48.000000000 -0400
@@ -256,9 +256,7 @@
 	};
 #endif
 
-#ifdef OPENSSL_SYS_SUNOS
-	typedef int sig_atomic_t;
-#endif
+typedef int sig_atomic_t;
 
 #if defined(OPENSSL_SYS_MACINTOSH_CLASSIC) || defined(MAC_OS_GUSI_SOURCE) || defined(OPENSSL_SYS_NETWARE)
 /*
